### ISSUE TITLE
[API DESIGN REVIEW] Keras loss function API refactor

### ISSUE THREAD

**Issue by yanboliang** [keras-team/keras#11224]
Refactor loss function from Python function to Python class.

See Keras API Design Review at https://docs.google.com/document/d/1FEmHbacBAxtmycI9Uta9FUBrqbPtXhqkLDwnOZRpCNE/edit?usp=sharing

I have a prototype which can verify it works well under the existing unit tests and integrated tests. If this get approved, I can submit the PR.

**Comment by pavithrasv** [comment]
Thank you for the design @yanboliang. We are currently working on a design for losses which is very close to what you have shared. I will provide a link to that in this thread as soon as it is ready. We can then look into the implementation logistics.
